app/controllers/DateOfBirthController.scala

```diff
@@ -16,57 +16,60 @@
-                                        val controllerComponents: MessagesControllerComponents,
-                                        view: DateOfBirthView
-                                      )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
-
-  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
-    implicit request =>
-
-      val form = formProvider()
+class DateOfBirthController @Inject()(sessionRepository: SessionRepository,
+                                      identify: IdentifierAction,
+                                      getData: DataRetrievalAction,
+                                      requireData: DataRequiredAction,
+                                      formProvider: DateOfBirthFormProvider,
+                                      view: DateOfBirthView,
+                                      registrationConnector: RegistrationConnector)
+                                     (implicit mcc: MessagesControllerComponents, ec: ExecutionContext)
+  extends FrontendController(mcc) with I18nSupport {
 
-      val preparedForm = request.userAnswers.get(DateOfBirthPage) match {
-        case None => form
-        case Some(value) => form.fill(value)
-      }
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
+    val preparedForm = request.userAnswers.get(DateOfBirthPage) match {
+      case None => formProvider()
+      case Some(value) => formProvider().fill(value)
+    }
 
-      Ok(view(preparedForm, mode))
+    Ok(view(preparedForm, mode))
   }
 
-  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
-    implicit request =>
-
-      val form = formProvider()
-
-      form.bindFromRequest().fold(
-        formWithErrors =>
-          Future.successful(BadRequest(view(formWithErrors, mode))),
+  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+    formProvider().bindFromRequest().fold(
+      formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
+      value =>
+        for {
+          updatedAnswers <- Future.fromTry(request.userAnswers.set(DateOfBirthPage, value))
+          registerResponse <- register(updatedAnswers)
+          fullAnswers = updatedAnswers.copy(registrationResponse = registerResponse)
+          _ <- sessionRepository.set(fullAnswers)
+        } yield Redirect(DateOfBirthPage.nextPage(mode, fullAnswers))
+    )
+  }
 
-        value =>
-          for {
-            updatedAnswers <- Future.fromTry(request.userAnswers.set(DateOfBirthPage, value))
-            _              <- sessionRepository.set(updatedAnswers)
-          } yield Redirect(DateOfBirthPage.nextPage(mode, updatedAnswers))
+  private def register(answers: UserAnswers)(implicit request: Request[_]): Future[Option[RegistrationResponse]] =
```

Comment:
This method name is not good as it does not call register when answers.get(HasNinoPage) returns value different than Some(true).
Either the method needs renaming, or the logic needs to change in the controller. You must be mindful that we are overriding always the registerResponse value and in this case for the wrong reason we are setting it to None. It works, but not good practice in my mind.